tools/log_collector/log_collector/log_collector.py

```diff
@@ -274,20 +335,59 @@ def filter_pods(pod_objects, job_objects):
     return filter_pod_objects
 
 
+# get_gv_list_by_group returns group_version matched for given group
+def get_gv_list_by_group(api_gv_list, group):
+    return [group_version for group_version in api_gv_list if group_version.startswith(group)]
+
 # get_gv_by_group returns group_version matched for given group
-def get_gv_by_group(api_gv_list, group):
-    matched_resources = [group_version for group_version in api_gv_list if group_version.startswith(group)]
-    if matched_resources:
-        return matched_resources[0]
+def get_gv_by_group(api_gv_list, group_name, isPreferredVersion = True):
+    for group in api_gv_list:
+        if group.name == group_name:
+            if isPreferredVersion:
+                return group.preferred_version.group_version
+            else:
+                return [version.group_version for version in group.versions]
     return ''
 
+# get_resource_gv_by_name returns resource object and gv for given resource name
+def get_resource_gv_by_name(resource_map, name):
+    gv_resource_map = dict()
+    for gv, resources in resource_map.items():
+        for resource in resources:
+            if resource.name == name:
+                gv_resource_map[gv] = resource
+                continue
+
+    return gv_resource_map
 
 # get_resource_by_name returns resource object for given resource name
 def get_resource_by_name(resources, name):
     matched_resources = [resource for resource in resources if resource.name == name]
     if matched_resources:
         return matched_resources[0]
 
+# check_container checks whether current and previous containers of a single container pod exits
+def check_container(pod):
+    current_container, previous_container = False, False
+    container_statuses = pod.get('status', {}).get('containerStatuses', [])
+    if container_statuses:
+        container_status = container_statuses[0]
```

Comment:
How do we handle multiple container scenario?
Also are we currently able to capture init container logs? We are using init containers in both of our deployments. So we need to be able to capture it's logs

Reply:
Fixed for multiple containers and init containers.